go/private/rules/binary.bzl

```diff
@@ -86,6 +89,42 @@ go_binary = rule(
 )
 """See go/core.rst#go_binary for full documentation."""
 
+go_tool_binary = rule(
+    _go_binary_impl,
+    attrs = {
+        "data": attr.label_list(
```

Comment:
`go_tool_binary` has a lot of attributes that can never be used. Maybe it's fine since this is an internal rule, but until this is merged with `go_binary`, maybe it makes more sense to delete these, and use `getattr` in the implementation or have separate implementations?

No need to change this if it doesn't make sense in the context of the rest of your changes. It just seems a little odd. Should be fine when optional toolchains lands and this can be cleaned up.

Reply:
I want to leave it identical to the attribute list on go_binary to make it easy to merge them one day.
I think that every parameter that should not be set will end up with an error at the compile or link action anyway.